db/tables.py

```diff
@@ -170,14 +170,54 @@ def _create_split_insert_stmt(
     return split_ins
 
 
+def merge_tables(
+        table_name_one,
+        table_name_two,
+        merged_table_name,
+        schema,
+        engine,
+        drop_original_tables=False,
+):
+    """
+    This specifically undoes the `extract_columns_from_table` (up to
+    unique rows).  It may not work in other contexts (yet).
+    """
+    t1 = reflect_table(table_name_one, schema, engine)
+    t2 = reflect_table(table_name_two, schema, engine, metadata=t1.metadata)
+    j = t1.join(t2)
+    referencing_columns = [
+        c for c in [j.onclause.left, j.onclause.right] if c.foreign_keys
+    ]
+    merged_columns_all = [
+        col.MathesarColumn.from_column(c)
+        for c in list(t1.columns) + list(t2.columns)
```

Comment:
I prefer to avoid single character variable names, could you make it more descriptive?

Reply:
Sure, I should have gone through what I was bringing over from the prototype a little more closely.